### PR TITLE
feat: add NativeOverlappedPresenter support for macOS/Skia host

### PR DESCRIPTION
**Pull request by spouliot**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

The new `NativeOverlappedPresenter` was **not** supported by the new macOS/Skia host.

## What is the new behavior?

The new `NativeOverlappedPresenter` is supported by the new macOS/Skia host.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Runtime.Skia.MacOS/UnoNativeMac/UnoNativeMac/UNOWindow.m [hunk 1]
```diff
@@ -165,11 +165,185 @@ void uno_window_set_min_size(NSWindow *window, double width, double height)
...
+    NSLog(@"uno_window_restore %@ %s", window, activateWindow ? "true" : "false");
+#endif
+    switch(uno_window_get_overlapped_presenter_state(window)) {
+        case OverlappedPresenterStateMaximized:
+            [window zoom:nil];
+            break;
+        case OverlappedPresenterStateMinimized:
+            [window deminiaturize:nil];
+            break;
+        default:
```
Comment (ajpinedam) [inline src/Uno.UI.Runtime.Skia.MacOS/UnoNativeMac/UnoNativeMac/UNOWindow.m:245]: Should we NSLog in situations where we hit this default case?

File: src/Uno.UI.Runtime.Skia.MacOS/UnoNativeMac/UnoNativeMac/UNOWindow.m [hunk 2]
```diff
@@ -165,11 +165,185 @@ void uno_window_set_min_size(NSWindow *window, double width, double height)
...
+#if DEBUG
+    NSWindowCollectionBehavior cb = window.collectionBehavior;
+#endif
+    // unlike Windows on macOS the (green) maximizable button is for full screen, not zoomed
+    // `windowShouldZoom:toFrame:` will check the collectionBehavior to [dis]allow zooming
+    if (isMaximizable) {
+        [window setCollectionBehavior:NSWindowCollectionBehaviorDefault];
+        [[window standardWindowButton:NSWindowZoomButton] setEnabled:YES];
+    } else {
+        [window setCollectionBehavior:NSWindowCollectionBehaviorFullScreenAuxiliary|NSWindowCollectionBehaviorFullScreenNone|NSWindowCollectionBehaviorFullScreenDisallowsTiling];
```
Comment (ajpinedam) [inline src/Uno.UI.Runtime.Skia.MacOS/UnoNativeMac/UnoNativeMac/UNOWindow.m:307]: oh Objective-C mon dieu! :D

File: src/Uno.UI.Runtime.Skia.MacOS/UnoNativeMac/UnoNativeMac/UNOWindow.m [hunk 3]
```diff
@@ -165,11 +165,185 @@ void uno_window_set_min_size(NSWindow *window, double width, double height)
...
+        style ^= NSWindowStyleMaskMiniaturizable;
+#if DEBUG
+    NSLog(@"uno_window_set_minimizable %@ 0x%x %s 0x%x", window, (uint)window.styleMask, isMinimizable ? "true" : "false", (uint)style);
+#endif
+    window.styleMask = style;
+}
+
+bool uno_window_set_modal(NSWindow *window, bool isModal)
+{
+    // this is a read-only property so we simply log if we can't change it o the requested value
```
Comment (ajpinedam) [inline src/Uno.UI.Runtime.Skia.MacOS/UnoNativeMac/UnoNativeMac/UNOWindow.m:330]: Based on this message, should we log something here?

**Review by ajpinedam — APPROVED** [review]

File: src/Uno.UI.Runtime.Skia.MacOS/UnoNativeMac/UnoNativeMac/UNOWindow.m (hunk 1 above)
Comment (spouliot) [reply src/Uno.UI.Runtime.Skia.MacOS/UnoNativeMac/UnoNativeMac/UNOWindow.m:245]: It was not useful to me 😄 

The `NSLog` are only present in `DEBUG` builds and the nuget only ship the release version of the binary. IOW the `NSLog` is only helpful if you build the native library yourself. Anything that needs to be logged is done from the managed assembly.

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**